### PR TITLE
kafka topic blacklist

### PR DESCRIPTION
**Pull request by yyejun**
Support kafka consumer blacklist.

### CODE REVIEW

File: src/main/java/com/pinterest/secor/reader/MessageReader.java [hunk 1]
```diff
@@ -65,7 +66,12 @@ public MessageReader(SecorConfig config, OffsetTracker offsetTracker) throws
 
         mConsumerConnector = Consumer.createJavaConsumerConnector(createConsumerConfig());
 
-        TopicFilter topicFilter = new Whitelist(mConfig.getKafkaTopicFilter());
+        if (!mConfig.getKafkaTopicBlacklist().isEmpty() && !mConfig.getKafkaTopicFilter().isEmpty()) {
+            throw new RuntimeException("Topic filter and blacklist cannot be both specified.");
+        }
+        TopicFilter topicFilter = !mConfig.getKafkaTopicBlacklist().isEmpty()? new Blacklist(mConfig.getKafkaTopicBlacklist()):
```
Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/reader/MessageReader.java]: Do you need to check both null and empty string?

Comment (yyejun) [inline src/main/java/com/pinterest/secor/reader/MessageReader.java]: If both empty, it will behavior same as current code return a WhileList(""). Not sure if it matches everything or nothing.

Comment (HenryCaiHaiying) [inline src/main/java/com/pinterest/secor/reader/MessageReader.java]: Actually I don't know whether you will get null from mConfig.getKafkaTopicBlacklist(), you probably will get NPE is the property is not specified.  Just do a test on not specifying that new property, if that looks good, I can merge in your PR.

Comment (yyejun) [inline src/main/java/com/pinterest/secor/reader/MessageReader.java]: If it is not specified instead of empty string, it will not pass https://github.com/pinterest/secor/blob/master/src/main/java/com/pinterest/secor/common/SecorConfig.java#L458.
I also tested both empty value, it runs but empty WhiteList.

**Comment by HenryCaiHaiying** [comment]
Sounds good, thanks for the enhancement.
